src/app/shared/fabric8-ui-http.service.ts

```diff
@@ -14,9 +15,43 @@ import { AUTH_API_URL, HttpService , SSO_API_URL } from 'ngx-login-client';
 import { Observable } from 'rxjs';
 import * as uuidV4 from 'uuid/v4';
 
+type PendingRequests = {
+  [cacheKey: string]: Observable<Response>;
+};
+
+function isGetMethod(url: string | Request, options?: RequestOptionsArgs): boolean {
+  return (typeof url === 'string' && (options == null || options.method === RequestMethod.Get)) ||
+    (url as Request).method === RequestMethod.Get;
+}
+
+// Creates a cache key from headers, query params, and URL
+function createCacheKey(req: string | Request, options?: RequestOptionsArgs): string {
+  let key;
+  if (typeof req === 'string') {
+    key = req;
+    if (options) {
+      if (options.params) {
+        key += `_${options.params.toString()}`;
+      }
+      if (options.headers) {
+        key += `_${JSON.stringify(options.headers.toJSON())}`;
+      }
+    }
+  } else {
+    key = req.url;
+    if (req.headers) {
+      key += `_${JSON.stringify(req.headers.toJSON())}`;
+    }
+  }
+  return key;
+}
+
+const TTL = 200; // cache requests for N milliseconds
```

Comment:
This seems like a reasonable timeout

Reply:
@dgutride have a look at the long comment in the code. Any thoughts on an expiring cache observable?